core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -690,6 +772,14 @@ class KafkaServer(
 
       _brokerState = BrokerState.PENDING_CONTROLLED_SHUTDOWN
 
+      if (config.migrationEnabled && lifecycleManager != null) {
+        // TODO KAFKA-14447 Only use KRaft controlled shutdown (when in migration mode)
```

Comment:
so the TODO here is that we need to check if the controller is a kraft one?

fair enough.

I keep thinking that info should be in the metadata cache ... I guess we can do that in a follow-up.

Reply:
Right, we'll read this information from MetadataCache once we get the UMR changes merged.